girder/api/v1/resource.py

```diff
@@ -66,30 +65,23 @@ def __init__(self):
         .errorResponse('Invalid type list format.')
     )
     def search(self, q, mode, types, level, limit, offset):
+        """This function return an empty dict if the mode isn't in the search mode
+        registry, else it call the search mode handler.
+        """
+        results = {}
         level = AccessType.validate(level)
         user = self.getCurrentUser()
+        handler = search_mode_utilities.getSearchModeHandler(mode)
 
-        if mode == 'text':
-            method = 'textSearch'
-        else:
-            method = 'prefixSearch'
-
-        results = {}
-        for modelName in types:
-            if modelName not in allowedSearchTypes:
-                continue
-
-            if '.' in modelName:
-                name, plugin = modelName.rsplit('.', 1)
-                model = self.model(name, plugin)
-            else:
-                model = self.model(modelName)
-
-            results[modelName] = [
-                model.filter(d, user) for d in getattr(model, method)(
-                    query=q, user=user, limit=limit, offset=offset, level=level)
-            ]
-
+        if handler is not None:
```

Comment:
If `handler` is None, I think we should raise a `RestException` (since the user-provided `mode` was invalid), rather than silently returning an empty dict.